libs/partners/huggingface/langchain_huggingface/llms/huggingface_endpoint.py

```diff
@@ -145,19 +145,18 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             )
 
         values["model_kwargs"] = extra
-
-        values["endpoint_url"] = get_from_dict_or_env(
-            values, "endpoint_url", "HF_INFERENCE_ENDPOINT", None
-        )
-
-        if values["endpoint_url"] is None and "repo_id" not in values:
+        if "endpoint_url" not in values and "repo_id" not in values:
```

Comment:
Could you update the doc-string for the wrapper to explain which variables can be specified?

----

I haven't used this wrapper, but the new code does not behave correctly as far as I can tell. If repo_id is not strictly necessary for initializaiton, then a user may be relying on endpoint_url being specified via an env variable. This scenario will cause a validation error right now.


> I am trying to initialize the HuggingFaceEndpoint, but despite passing the correct repo_id, I am encountering an error. I have identified the bug: even though I provide the repo_id, the HuggingFaceEndpoint validation always checks for the endpoint_url, which is incorrect. If the repo_id is passed, it should not be checking for the endpoint_url. I will create a PR to fix this issue.

Old validation does not seem to be checking endpoint_url, it seems to be *loading* it from the environment by default.

Possible solutions:
1) Throw an error if both values can be loaded
2) Only try to load_endpoint url after repo_id is missing and only then do the check that at least one of them is specified (this is still too implicit -- so bad solution)
3) Change precedence order if it's obvious that repo_id should be favored over endpoint_url (too implicit -- so bad solution)

Reply:
- I have updated the wrapper's docstring.

- Previously, the `endpoint_url` was first checked from environment variables. If the `endpoint_url` was not set as an environment variable and only `repo_id` was provided, an error would occur despite the `repo_id` being passed. I have now revised this behavior so that the code first checks whether the user has provided a `repo_id` or `endpoint_url` directly. Only after that, if `repo_id` is not provided will it then retrieve the values from the environment variables.

-  See below please.
```python
llm = HuggingFaceEndpoint(
                repo_id="microsoft/Phi-3-mini-4k-instruct", # user intend to use repo_id but validation error will be raised
for `endpoint_url` env variable missing which is wrong since user has passed repo_id
                huggingfacehub_api_token = 'xxxxxx',
                task="text-generation",
                max_new_tokens=512,
                do_sample=False,
                repetition_penalty=1.03,
            )
```